index.js

```diff
@@ -49,26 +49,27 @@ function fresh (reqHeaders, resHeaders) {
   }
 
   // if-none-match
-  if (noneMatch && noneMatch !== '*') {
+  if (noneMatch) {
+    if (noneMatch === '*') {
+      return true
+    }
     var etag = resHeaders.etag
 
     if (!etag) {
       return false
     }
 
-    var etagStale = true
+    var etagStale = false
```

Comment:
Does this name make sense now? Shouldn't it be changed to `etagFresh`?

Reply:
Agreed, it has been modified.